### PR TITLE
Implement more flags for 'restic forget'

### PR DESCRIPTION
**Pull request by baurmatt**
Moving forward with this project, it might be worth considering to dynamically allow all possible parameters for restic instead of statically implement each and everyone. But that will be a lot of refactoring and breaking changes :)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -108,8 +108,13 @@ snapshot 3671204c saved
 ### Args
 
 * `prune`: A boolean representing whether to automatically run the 'prune' command if snapshots have been removed
+* `keep_last`: An int representing the last N snapshots to keep
+* `keep_hourly`: An int representing the last N hourly snapshots to keep
 * `keep_daily`: An int representing the last N daily snapshots to keep
 * `group_by`: A string for grouping snapshots
+* `keep_weekly`: An int representing the last N weekly snapshots to keep
+* `keep_monthly`: An int representing the last N monthly snapshots to keep
+* `keep_yearly`: An in representing the last N yearly snapshots to keep
```
Comment (mtlynch) [inline docs/index.md:117]: `in` -> `int`

File: docs/index.md [hunk 2]
```diff
@@ -108,8 +108,13 @@ snapshot 3671204c saved
 ### Args
 
 * `prune`: A boolean representing whether to automatically run the 'prune' command if snapshots have been removed
+* `keep_last`: An int representing the last N snapshots to keep
+* `keep_hourly`: An int representing the last N hourly snapshots to keep
 * `keep_daily`: An int representing the last N daily snapshots to keep
 * `group_by`: A string for grouping snapshots
```
Comment (mtlynch) [inline docs/index.md:114]: Can we move `group_by` either above or below the `keep_*` so that all the `keep_` args are contiguous?

**Review by mtlynch — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 1 above)
Comment (baurmatt) [reply docs/index.md:117]: Good catch, thanks! :)

File: docs/index.md (hunk 2 above)
Comment (baurmatt) [reply docs/index.md:114]: Yep, that makes sense! Done! :)

**Comment by baurmatt** [comment]
I've added two more changes:

* Implement dry-run for 'restic forget'
* Sort forget parameters alphabetically

Hope it's fine to just "mix" it into this PR.

**Comment by mtlynch** [comment]
>Hope it's fine to just "mix" it into this PR.

It's okay on this one, but try to keep the PRs narrowly scoped and avoid adding scope after a review starts. I've written up guidelines to get your PRs merged as quickly as possible here:

https://mtlynch.io/code-review-love/

> baurmatt force-pushed the forget_keep branch from cb44cd8 to 45b874c now 

Can you try to avoid force-pushing? It makes it hard to review your changes. I merge the PRs with squash and merge, so all the commits get squashed into a single one on merge.

**Comment by mtlynch** [comment]
It looks like these changes are breaking the build:

https://app.circleci.com/pipelines/github/mtlynch/resticpy/226/workflows/722eb347-ba7b-4bdf-8053-3fafb96506f8/jobs/494

You can run these tests locally with `./dev-scripts/build`, and you can set them to run automatically on every commit with `./dev-scripts/hooks/enable-hooks`.

**Comment by baurmatt** [comment]
Thanks! :) I sometimes just miss running tests before pushing 🙈 Automatically running them on commit doesn't work because I have global hooks defined in `~/.gitconfig`:

```
[core]
  hooksPath = /Users/mbaur/.git-hooks
```

Howevery.... I've fixed all of those bugs the linting/testing complained about, just one is missing where I don't have a good idea for it:

```
restic/internal/forget.py:7:0: R0913: Too many arguments (11/10) (too-many-arguments)
```

I guess it's just what I've mentioned in the PR description.

**Comment by mtlynch** [comment]
>I guess it's just what I've mentioned in the PR description.

I've just added #87 to fix that, although I'm open to a dynamic approach if you want to open an issue sharing a sketch of how that would work.

**Review by mtlynch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**